crowbar_framework/lib/upgrade.rb

```diff
@@ -0,0 +1,58 @@
+class Upgrade
+  class << self
+    # Return the current state of upgrade process.
+    # We're keeping the information in the file so is accessible by other applications
+    # and different crowbar versions.
+    def progress
+      upgrade_progress = {
+        current_step: "prechecks",
```

Comment:
I guess it should be `upgrade_prechecks`, or at least the same as in the list of the steps, so `current_step: upgrade_steps_6_7.first` should be better here i guess